cubi-tools/prototypes/update_workflow.py

```diff
@@ -0,0 +1,646 @@
+                        f"version '{target_metadata_version}' to version "
+                        f"'{branch_metadata_version}' and "
+                        "The 'pyproject.toml' workflow version was NOT updated from "
+                        f"version '{target_workflow_version}' to version "
+                        f"'{branch_workflow_version}'!"
+                    )
+        else:
+            print(
+                "\nMetadata and workflow versions in 'pyproject.toml' are "
+                "up-to-date!\n"
+            )
+    return None
+
+
+def user_response(question, attempt=0):
+    """
+    Function to evaluate the user response to the Yes or No question refarding updating
+    the metadata files.
+    """
+    attempt += 1
+    prompt = f"{question}? (y/n): "
+    answer = input(prompt).strip().lower()
+    pos = ["yes", "y", "yay"]
+    neg = ["no", "n", "nay"]
+    if attempt == 2:
+        print("YOU HAVE ONE LAST CHANCE TO ANSWER THIS (y/n) QUESTION!")
+    if attempt >= 3:
+        raise RuntimeError(
+            "I warned you! You failed 3 times to answer a simple (y/n)"
+            " question! Please start over!"
+        )
+    if not (answer in pos or answer in neg):
+        print(f"That was a yes or no question, but you answered: {answer}")
+        return user_response(question, attempt)
+    return answer in pos
+
+
+def find_all_subdir_in_branch(workflow_branch):
+    """
+    Function to list all subdirectories in the 'template_snakemake' repo
+    to be able to create all missing subfolders.
+    """
+    subfolders = [
+        branch_folder.path
+        for branch_folder in os.scandir(workflow_branch)
+        if branch_folder.is_dir()
+    ]
+    for workflow_branch in list(subfolders):
+        subfolders.extend(find_all_subdir_in_branch(workflow_branch))
+    return subfolders
+
+
+def update_file_list(workflow_branch, metadata, dryrun):
+    """
+    Function to create a list of files that will be updated.
+    """
+    files_to_update = []
+    # create a list of all files in 'template_snakemake' directory
+    workflow_files = []
+    for file in workflow_branch.rglob("*"):
+        if file.is_file():
+            workflow_files.append(str(pathlib.Path(file).relative_to(workflow_branch)))
+
+    # the following files need to be excluded because they are always project specific
+    excluded_files = [
+        "pyproject.toml",
+        "workflow/rules/00_modules.smk",
+        "workflow/rules/99_aggregate.smk",
+    ]
+    workflow_files = [item for item in workflow_files if item not in excluded_files]
+    # the '.git' folder also needs to be excluded from the update list!
+    workflow_files = [item for item in workflow_files if ".git/" not in item]
+    # I want the 'pyproject.toml' file at the end of the list and therefore I excluded
+    # it initially from the list and now I append it so it will be at the end
+    workflow_files.append("pyproject.toml")
+
+    subfolders = [
```

Comment:
same as above --- and:

obviously, you need a function to just return a list of subdir or something like that (you are repeating yourself here)

Reply:
This function already exists, see above. I just forgot to call it.